datastore_orm/model.py

```diff
@@ -406,13 +406,14 @@ def background_delete(self, key, clients):
     def delete(self):
         """Delete object from datastore.
         """
-        self._clients[0].delete(self)
+        if self._cache:
+            cache_key = 'datastore_orm.{}.{}'.format(self.kind, self.id_or_name)
+            self._cache.delete(cache_key)
+
         if len(self._clients) > 1:
             Thread(target=self.background_delete, args=(self, self._clients[1:])).start()
 
-        if self._cache:
-            cache_key = 'datastore_orm.{}.{}'.format(self.__class__.__name__, self.key.id_or_name)
-            self._cache.delete(cache_key)
+        self._clients[0].delete(self)
```

Comment:
run this first before triggering the background delete. primary update should finish before we attempt any secondary updates.